### PR TITLE
t1lib: add test

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/t/t1lib.rb [hunk 1]
```diff
@@ -39,4 +39,56 @@ def install
...
+    EOS
+
+    expected_output = <<~EOS
+      FontID=0, Font=./bchr.pfb
+      FontID=0, Metrics=./bchr.afm
+      Dataformat: T1_bit=0, T1_byte=0, T1_wordsize=16, T1_pad=16
+    EOS
+
+    with_env(T1LIB_CONFIG: testpath/"t1lib.config") do
+      assert_equal expected_output, shell_output("./test")
```
Comment (SMillerDev) [inline Formula/t/t1lib.rb:91]: It seems like the Linux one also prints some ascii art underneath.
```suggestion
      assert_match expected_output, shell_output("./test")
```

Comment (alebcay) [reply Formula/t/t1lib.rb:91]: I think it's supposed to be printed/dumped in all platforms. Trying to figure out why it only seems to be happening on Linux,

**Comment by alebcay** [comment]
Still investigating, but looks like the rasterization code might be broken on macOS? The following error is logged with some debug statements when running in macOS:

```
(E) T1_SetString(): t1_abort: Reason: unable to fix subpath break?
```

This seems to be emitted from [here (line 543)](https://fossies.org/linux/t1lib/lib/t1lib/t1set.c). Actual error is being encountered further than that.

File: Formula/t/t1lib.rb [hunk 2]
```diff
@@ -39,4 +39,55 @@ def install
...
+        for( i=0; i<T1_GetNoFonts(); i++){
+          printf("FontID=%d, Font=%s\\n", i, T1_GetFontFilePath(i));
+          printf("FontID=%d, Metrics=%s\\n", i, T1_GetAfmFilePath(i));
+          // T1_DumpGlyph(T1_SetString( i, "Test", 0, 0, T1_KERNING, 25.0, NULL));
+        }
+
+        T1_CloseLib();
+        return EXIT_SUCCESS;
+      }
+    EOS
```
Comment (carlocab) [inline Formula/t/t1lib.rb:70]:
```suggestion
    (testpath/"test.c").write <<~C
      #include <stdio.h>
      #include <stdlib.h>
      #include <t1lib.h>

      int main( void)
      {
        int i;
        T1_SetBitmapPad(16);

        if ((T1_InitLib(NO_LOGFILE)==NULL)){
          fprintf(stderr, "Initialization of t1lib failed\\n");
          return EXIT_FAILURE;
        }

        for( i=0; i<T1_GetNoFonts(); i++){
          printf("FontID=%d, Font=%s\\n", i, T1_GetFontFilePath(i));
          printf("FontID=%d, Metrics=%s\\n", i, T1_GetAfmFilePath(i));
          // T1_DumpGlyph(T1_SetString( i, "Test", 0, 0, T1_KERNING, 25.0, NULL));
        }

        T1_CloseLib();
        return EXIT_SUCCESS;
      }
    C
```
This makes some editors (and I think maybe GitHub now) do the correct syntax highlighting here.

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**